LEGO1/lego/legoomni/src/act1/act1state.cpp

```diff
@@ -1,34 +1,287 @@
 #include "act1state.h"
 
+DECOMP_SIZE_ASSERT(Act1State, 0x26c)
+DECOMP_SIZE_ASSERT(Act1State::NamedPlane, 0x4c)
+DECOMP_SIZE_ASSERT(Act1State::NamedTexture, 0x14)
+
+// Forward declarations
+Act1State::NamedTexture* ReadNamedTexture(LegoFile* p_file);
+void WriteNamedTexture(LegoFile* p_file, Act1State::NamedTexture* p_texture);
+void FUN_1003f540(LegoFile* p_file, const char* p_filename);
+
+// GLOBAL: ISLE 0x100f37f0
+MxS32 g_unk0x100f37f0[] = {
+	Act1State::e_unk953,
+	Act1State::e_unk954,
+	Act1State::e_unk955,
+};
+
 // STUB: LEGO1 0x100334b0
-Act1State::Act1State()
+Act1State::Act1State() : m_unk0x00c(0), m_unk0x00e(0), m_unk0x008(NULL), m_unk0x010(0)
 {
-	// TODO
-	m_unk0x1e = 0;
-	m_unk0x18 = 1;
-	m_unk0x20 = 0;
-	m_unk0x1f = 0;
-	m_unk0x21 = TRUE;
-	m_unk0x22 = 0;
-	m_unk0x1c = 1;
+	m_unk0x01e = 0;
+	m_unk0x018 = 1;
+	m_unk0x010 = 0;
+	m_unk0x020 = 0;
+	m_unk0x00e = 0;
+	m_unk0x01f = 0;
+	m_unk0x008 = g_unk0x100f37f0;
+	m_unk0x014 = -1;
+	m_unk0x022 = 0;
+	m_unk0x154 = NULL;
+	m_unk0x158 = NULL;
+	m_unk0x15c = NULL;
+	m_unk0x160 = NULL;
+	m_unk0x1b0 = NULL;
+	m_unk0x021 = 1;
+	m_unk0x01c = 1;
+	m_unk0x00c = _countof(g_unk0x100f37f0);
+	m_unk0x1b4 = NULL;
+	m_unk0x1b8 = NULL;
+	m_unk0x208 = NULL;
+	m_unk0x20c = NULL;
+	m_unk0x25c = NULL;
+	m_unk0x260 = NULL;
+	m_unk0x264 = NULL;
+	m_unk0x268 = NULL;
+	SetFlag();
 }
 
-// STUB: LEGO1 0x10033ac0
+// FUNCTION: LEGO1 0x10033ac0
 MxResult Act1State::VTable0x1c(LegoFile* p_legoFile)
 {
+	if (p_legoFile->IsWriteMode()) {
+		p_legoFile->FUN_10006030(ClassName());
+	}
+	m_unk0x024.Serialize(p_legoFile);
```

Comment:
This function appears to be inlined here at least a few times - so the `Serialize` function should be implemented in the header or at least declared inline (although we generally prefer having the implementation located in the header)